receiver/jmxreceiver/config.go

```diff
@@ -167,12 +174,64 @@ func (c *Config) parseClasspath() string {
 	return strings.Join(classPathElems, ":")
 }
 
+func hashFile(path string) (string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	hasher := sha256.New()
+	if _, err := io.Copy(hasher, f); err != nil {
+		return "", err
+	}
+
+	return hex.EncodeToString(hasher.Sum(nil)), nil
```

Comment:
Mostly readability & the custom hash functionality. They could certainly be left as a byte slice/array but the supported_jars file also felt easier to parse & validate as a developer when they are strings.

Reply:
I suppose there are potential reasons to stay closer to the sha256 representation.
1. Encoding as hex may be meaningfully reducing the space of unique values, such that it would be easier to spoof a jar. Realistically, this is still very difficult, but perhaps much more crackable by brute force.
2. Possibly it is more difficult to hash a jar for a custom build, if it must be run through both sha256 and hexed.

Reply:
1. I don't believe there is any reduction in cardinality by translating bytes to hex, this is just a shorter representation of the same data.

Reply:
Very true. I had a bug in my thought process there.